### ISSUE TITLE
Why use multiple classes for Langchain and OpenAI 

### ISSUE THREAD

**Issue by krrishdholakia** [MaartenGr/BERTopic#1452]
Hi @MaartenGr, 

 interesting approach to abstracting llm providers - by creating separate classes langchain.py and openai.py

Why have them as 2 separate classes? doesn't langchain cover the openai scenario already?

**Comment by MaartenGr** [comment]
In a way, it definitely does! However, there have been users that prefer control over OpenAI directly without any of the interfacing of LangChain. Moreover, the OpenAI scenario is dependency-wise significantly smaller and with fewer dependency issues than a large package such as LangChain resulting in much easier deployment. Dependency hell is definitely something not to be underestimated.

**Comment by krrishdholakia** [comment]
@MaartenGr agreed on the bloat. Curious - what kind of things about openai are you trying to control that you can't with langchain? I don't think i caught that when i was looking at the files

**Comment by MaartenGr** [comment]
It is not necessarily that OpenAI does more than Langchain but indeed the additional bloat that comes with Langchain.
